staging/src/k8s.io/kubectl/pkg/cmd/create/create_job.go

```diff
@@ -254,29 +256,33 @@ func (o *CreateJobOptions) createJob() *batchv1.Job {
 	return job
 }
 
-func (o *CreateJobOptions) createJobFromCronJobV1Beta1(cronJob *batchv1beta1.CronJob) *batchv1.Job {
+func (o *CreateJobOptions) createJobFromCronJobV1Beta1(cronJob *batchv1beta1.CronJob, controller bool) *batchv1.Job {
 	annotations := make(map[string]string)
 	annotations["cronjob.kubernetes.io/instantiate"] = "manual"
 	for k, v := range cronJob.Spec.JobTemplate.Annotations {
 		annotations[k] = v
 	}
 
+	ownerReference := metav1.OwnerReference{
+		// TODO (soltysh): switch this to v1 in v1.22, when n-1 skew will be fulfilled
+		APIVersion: batchv1beta1.SchemeGroupVersion.String(),
+		Kind:       "CronJob",
+		Name:       cronJob.GetName(),
+		UID:        cronJob.GetUID(),
+	}
+	if controller {
+		ownerReference.Controller = &controller
+		ownerReference.BlockOwnerDeletion = &controller
+	}
```

Comment:
Here is a quick question: if controller is a 'must' when --from is defined, wouldn't be better to verify if the len of o.From is > 0 and then set o.Controller = true? 

Is there a case where --controller is not desired, when setting --from?

If not, and the correct behavior is to always set the controller field, this could be part of the Complete() function, setting the bool to true if len(o.From) > 0 :)

OR you can simply verify inside the functions if len(o.From) > 0, and then set the other fields in ownerReference.Controller :)

Reply:
Originally the default `o.Controller=true`, but @soltysh thinks this is a bug and has been merged #91053. 
This is an incompatible modification, some users rely on this.

Reply:
ah ok, I see :D thanks for the explanation